src/core/challenge_api.py

```
@@ -1,23 +1,108 @@
+
+    def __str__(self):
+        return self.id
+
+    def save(self):
+        json_data = dict(id=self.id, features=self.features)
+        if not os.path.isdir(self.folder_path):
+            os.mkdir(self.folder_path)
+        if not os.path.isfile(self.file_path):
+            out_file = open(f"{self.file_path}", "w")
+            json.dump(json_data, out_file)
+            out_file.close()
+
+    def get(self):
+        try:
+            file_json = open(f"{self.file_path}", "r")
+            data_json = json.load(file_json)
+            file_json.close()
+            return data_json
+        except FileNotFoundError:
+            print("Error: The file does not exist")
+
+    def delete(self):
+        print(f"Deleting vehicle {self.id} of user: {self.user_id}, are you sure ? YES/no")
+        if input() == "YES":
+            try:
+                os.remove(self.file_path)
+                print("Vehicle deleted successfully")
+            except FileNotFoundError:
+                print("Error: The file does not exist")
+        else:
+            print("Aborted")
+
+
 @api.route("/challenge", methods=["POST"])
 @ApiDecorators.require_customer_id
-def vehicle_features_post(user_id: str):
-    """
-    Please see the README.md
-    Also see `json/vehicle-features.v1.schema.json` and `json/vehicle-features.v1.example.json`
+def vehicle_features_post(user_id):
+    vehicles = clean_input_data()  # format input data
+    if is_valid(vehicles):  # validate JSON schema
+        try:
+            saving_vehicles(vehicles, user_id)  # save vehicle to JSON
+            return make_response("200", 200)
+        except FileExistsError:
+            return make_response("400", 400)
+        except:
+            return make_response("500", 500)
+    else:
+        return make_response("400", 400)
+
+
+def clean_input_data():
+    input_data = dict()
+    try:
+        input_data = request.get_json()
+    except:
```

You do not want to use catch all exception handlings in general.